### ISSUE TITLE
Missing initialization of `series_index` when using OECD agency

### ISSUE THREAD

**Issue by rudolf-bauer** [dr-leo/pandaSDMX#64]
Hi @dr-leo, 

when using `pandasdmx` for a download of the OECD data, I am getting the following error.

To reproduce:
```
from pandasdmx import Request
symbol = "SNA_TABLE1/AUS+AUT+BEL+CAN+CZE+DNK+FIN+FRA+DEU+GRC+HUN+ISL+IRL+ITA+JPN+KOR+LUX+MEX+NLD+NZL+NOR+POL+PRT+SVK+ESP+SWE+CHE+TUR+GBR+USA+ARG+NMEC+BRA+BGR+CHL+CHN+COL+CRI+HRV+CYP+EST+IND+IDN+ISR+LVA+LTU+MKD+MLT+PER+ROU+RUS+SAU+SVN+ZAF+DEW+FRME+OTF+EU15+EU28+OECD+OECDE+EA19.B1_GA.C"
request = Request(agency='OECD', log_level=10)
response = request.data(symbol)
df = response.write()
```

The error:
```
UnboundLocalError                         Traceback (most recent call last)
<ipython-input-45-2617ddf5597f> in <module>()
      3 request = Request(agency='OECD', log_level=10)
      4 response = request.data(symbol)
----> 5 df = response.write()

/opt/virtualenv/develop/lib/python3.5/site-packages/pandasdmx/api.py in write(self, source, **kwargs)
    633         if not source:
    634             source = self.msg
--> 635         return self._writer.write(source=source, **kwargs)
    636 
    637     def write_source(self, filename):

/opt/virtualenv/develop/lib/python3.5/site-packages/pandasdmx/writer/data2pandas.py in write(self, source, asframe, dtype, attributes, reverse_obs, fromfreq, parse_time)
    107                 series_list = list(s for s in self.iter_pd_series(
    108                     iter_series, dim_at_obs, dtype, attributes,
--> 109                     reverse_obs, fromfreq, parse_time))
    110                 if dtype and attributes:
    111                     # series_list is actually a list of pairs of series

/opt/virtualenv/develop/lib/python3.5/site-packages/pandasdmx/writer/data2pandas.py in <genexpr>(.0)
    105         else:
    106             if asframe:
--> 107                 series_list = list(s for s in self.iter_pd_series(
    108                     iter_series, dim_at_obs, dtype, attributes,
    109                     reverse_obs, fromfreq, parse_time))

/opt/virtualenv/develop/lib/python3.5/site-packages/pandasdmx/writer/data2pandas.py in iter_pd_series(self, iter_series, dim_at_obs, dtype, attributes, reverse_obs, fromfreq, parse_time)
    226             if dtype:
    227                 value_series = PD.Series(
--> 228                     obs_values, index=series_index, name=series.key)
    229 
    230             if attributes:

UnboundLocalError: local variable 'series_index' referenced before assignment
```

I'm using the `master` branch. The [url](http://stats.oecd.org/SDMX-JSON/data/SNA_TABLE1/AUS+AUT+BEL+CAN+CZE+DNK+FIN+FRA+DEU+GRC+HUN+ISL+IRL+ITA+JPN+KOR+LUX+MEX+NLD+NZL+NOR+POL+PRT+SVK+ESP+SWE+CHE+TUR+GBR+USA+ARG+NMEC+BRA+BGR+CHL+CHN+COL+CRI+HRV+CYP+EST+IND+IDN+ISR+LVA+LTU+MKD+MLT+PER+ROU+RUS+SAU+SVN+ZAF+DEW+FRME+OTF+EU15+EU28+OECD+OECDE+EA19.B1_GA.C) is produced correctly and the data seems to be available in the response. However, the parsing in the writer fails.

If I understand it correctly, in lines [163-167](https://github.com/dr-leo/pandaSDMX/blob/master/pandasdmx/writer/data2pandas.py#L163) the frequency key should be detected. However, the response data has no frequency in the OECD data, but there is no else case for that. Specifically, the `if` in line [169](https://github.com/dr-leo/pandaSDMX/blob/master/pandasdmx/writer/data2pandas.py#L169) has no `else`. Therefore, the variable `series_index` is still not set in line [228](https://github.com/dr-leo/pandaSDMX/blob/master/pandasdmx/writer/data2pandas.py#L228). In my understanding, the simplest solution would be to add an else case after line [214](https://github.com/dr-leo/pandaSDMX/blob/master/pandasdmx/writer/data2pandas.py#L214) in order to close the `if` in line [169](https://github.com/dr-leo/pandaSDMX/blob/master/pandasdmx/writer/data2pandas.py#L169). Since there is no frequency available, one could just create a simple Index:
```
else:
    series_index = PD.Index(obs_dim, name=dim_at_obs)
```
I tried in locally and it seems to work.

Regards Rudi

**Comment by dr-leo** [comment]
Hi Rudi,

thank you for your thorough analysis and proposal to fix this. While I wasn't able to reproduce the bug due to some http error (I didn't get the response), I leave this issue open for later. Your proposal seems to make sense provided the series contained in the dataset really does not contain a freq dimension or attribute. 

Do you still encounter this bug? I have rewritten large parts of the relevant code for the forthcoming v0.9. - Reproducing the bug might be easier with less countries. I will retry in a few days.

Thanks again and best regards

Leo

**Comment by dr-leo** [comment]
I cannot reproduce the error due to some HTTP client error 400. But I've essentially done what you've suggested. So this should be fixed in v0.9 to be released very soon.

Thanks for spotting this.

Leo
